mfs/ops.go

```diff
@@ -12,60 +13,207 @@ import (
 	ipld "gx/ipfs/QmaA8GkXUYinkkndvg7T6Tx7gYXemhxjaxLisEPes7Rf1P/go-ipld-format"
 )
 
+var (
+	errInvalidNodeType            = errors.New("invalid argument")
+	errMvDirToFile                = errors.New("can not move directory to file path")
+	errMvParentDir                = errors.New("can not move parent directory to sub directory")
+	errInvalidDirPath             = errors.New("path end with '/' is not a directory")
+	errUnableCreateNonExistedFile = errors.New("unable to create un-existed file")
+)
+
 // Mv moves the file or directory at 'src' to 'dst'
 func Mv(r *Root, src, dst string) error {
-	srcDir, srcFname := gopath.Split(src)
+	if src == "/" {
+		return errMvParentDir
+	}
 
-	var dstDirStr string
-	var filename string
-	if dst[len(dst)-1] == '/' {
-		dstDirStr = dst
-		filename = srcFname
+	trimmedSrc := strings.TrimRight(src, "/")
```

Comment:
Is this to catch `mv /a/b/c/ /a/b/c`? If so, I'd do that separately and return a different error (to avoid any confusion).